src/map/packets/shop_items.cpp

```diff
@@ -51,6 +51,8 @@ CShopItemsPacket::CShopItemsPacket(CCharEntity * PChar)
 		ref<uint32>(i * 12 + 0x08) = PChar->Container->getQuantity(slotID);
 		ref<uint16>(i * 12 + 0x0C) = PChar->Container->getItemID(slotID);
 		ref<uint8>(i * 12 + 0x0E) = slotID;
+        ref<uint8>(i * 12 + 0x10) = PChar->Container->getGuildID(slotID);
+        ref<uint16>(i * 12 + 0x12) = PChar->Container->getGuildRank(slotID);
```

Comment:
Perform the math in the outgoing packet here instead. We don't know retail internally represents ranks with values of 100, 200, 300 based on just this packet (but if there's other indications, that's a different story). So long as we're passing the term "guild rank" around core, should probably be kept consistent across different uses (ie: a standard enum of 1-16).

If don't want to do the math each iteration: can do and set it in the binding, but just rename the "guildRank" in the container to something like "skillCap". That way it's clear that it's not to be confused with the 1-16 ranks, and is still an accurate descriptor of what the value might represent.

Reply:
Adjusted it to be done here at the packet.

One calculation per trade item isn't much so I'm not concerned there, and if we're going to push the rank metaphor down one layer to the variable, may as well push it all the way down to the packet.